### PR TITLE
Dto hotfixes & Nuke Intergration

### PR DESCRIPTION
**Pull request by AleXLaeR**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/settings.ruleset [hunk 1]
```diff
@@ -60,7 +60,7 @@
     <!-- A constant field is placed beneath a non-constant field. -->
     <Rule Id="SA1208" Action="Error"/>
     <!-- Using directive for 'System' must appear before -->
-    <Rule Id="SA1000" Action="Error" />
+    <Rule Id="SA1000" Action="None" />
```
Comment (dimaln) [inline Streetcode/settings.ruleset:63]: are we sure to ignore it? maybe use a warning or info message?

File: build/Targets/DataBaseBuild.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace Targets;
+
+partial class Build
+{
+    [Parameter("Specifies migration name during its creation")]
+    readonly string MigrName = "New Migration Added";
+     
+    Target AddMigration => _ => _
+        //ToDo add condition to check if there is any changes in ef code first
```
Comment (dimaln) [inline build/Targets/DataBaseBuild.cs:16]: Up to you, better create ticket for it and remove comment from the code

File: build/Targets/GitBuild.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    bool Checkouth = false;
+
+    [Parameter("name of branch to checkout")]
+    readonly string BName = "master";
+
+    [Parameter("create a new branch")]
+    bool NewB = false;
+
+    //[Parameter("allow to update submodules")]
+    //public bool WithCli { get => WithCli = true; }
```
Comment (dimaln) [inline build/Targets/GitBuild.cs:26]: remove

File: .docker/.env [hunk 4]
```diff
@@ -0,0 +1,4 @@
+﻿MS_SQL_DB_PORT=1433
+SA_PASSWORD=Sashacool1
+DB_USER=sa
+DB_NAME=StreetcodeDb
```
Comment (dimaln) [inline .docker/.env:4]: db named in CammelCase - StreetCodeDb
pls change name

File: Streetcode/Streetcode.WebApi/Program.cs [hunk 5]
```diff
@@ -31,4 +31,5 @@
 app.MapControllers();
 
 // await dbTask;
+//
```
Comment (dimaln) [inline Streetcode/Streetcode.WebApi/Program.cs:34]: remove

File: Streetcode/Streetcode.WebApi/Properties/launchSettings.json [hunk 6]
```diff
@@ -8,7 +8,7 @@
         "ASPNETCORE_ENVIRONMENT": "Development"
       },
       "dotnetRunMessages": true,
-      "applicationUrl": "https://localhost:7051;http://localhost:5258"
+      "applicationUrl": "https://localhost:5001;http://localhost:5000"
```
Comment (dimaln) [inline Streetcode/Streetcode.WebApi/Properties/launchSettings.json:11]: also need to update Readme.md

**Review by dimaln — CHANGES_REQUESTED** [review]

File: .docker/.env (hunk 4 above)
Comment (LanchevychMaxym) [inline .docker/.env:4]: You should create at least 2 databases(Streetcode and Streetcode_IntegrationTests)
My suggestion is to create nuke task CreateDatabase that creates all needed databases with simple sql, in that case you can use SA creds in each connection string

File: .docker/.env [hunk 7]
```diff
@@ -0,0 +1,4 @@
+﻿MS_SQL_DB_PORT=1433
+SA_PASSWORD=Sashacool1
```
Comment (LanchevychMaxym) [inline .docker/.env:2]: Let's consider some random password

File: Streetcode/.healthchecks/redis.sh [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+# shellcheck disable=SC2039
+set -eo pipefail
+
+host="$(hostname -i || echo '127.0.0.1')"
+
+if ping="$(redis-cli -h "$host" ping)" && [ "$ping" = 'PONG' ]; then
+	exit 0
+fi
+
+exit 1
```
Comment (LanchevychMaxym) [inline Streetcode/.healthchecks/redis.sh:11]: Do we need redis?

**Review by LanchevychMaxym — CHANGES_REQUESTED** [review]

File: build/Targets/BaseBuild.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
+﻿using Nuke.Common;
+using Nuke.Common.IO;
+using Nuke.Common.Tools.DotNet;
+using Nuke.Common.Utilities.Collections;
+using static Nuke.Common.IO.FileSystemTasks;
+using static Nuke.Common.Tools.DotNet.DotNetTasks;
+
+namespace Targets;
+
+partial class Build
```
[the remark this answers is not in the document]
Comment (AleXLaeR) [reply build/Targets/BaseBuild.cs:13]: We need to delegate different logic to separate parts of the same class, following the SRP principle.
Furthermore, we were following Nuke "best practices":
https://dev.to/damikun/the-cross-platform-build-automation-with-nuke-1kmc (here the author aslo uses partial class approach).

File: build/Targets/PublishBuild.cs [hunk 10]
```diff
@@ -0,0 +1,22 @@
+﻿using Nuke.Common;
+using static Nuke.Common.Tools.Git.GitTasks;
+
+namespace Targets;
+
+partial class Build
+{
+    Target PublishBackEnd => _ => _
```
[the remark this answers is not in the document]
Comment (AleXLaeR) [reply build/Targets/PublishBuild.cs:8]: Yes, we are.

File: build/Targets/FinalizerBuild.cs [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+using static Nuke.Common.Tools.PowerShell.PowerShellTasks;
+
+namespace Targets;
+
+partial class Build
+{
+    Target RemoveEnvVariables => _ => _
+        .Executes(() =>
+        {
+            PowerShell(@"reg delete HKEY_CURRENT_USER\Environment /v DOCKER_ATOM /f");
```
[the remark this answers is not in the document]
Comment (AleXLaeR) [reply build/Targets/FinalizerBuild.cs:13]: For the future.

File: .docker/.env (hunk 4 above)
Comment (AleXLaeR) [reply .docker/.env:4]: We are leaving it for the future scaling.

File: .docker/.env (hunk 7 above)
Comment (AleXLaeR) [reply .docker/.env:2]: We have set the default password for "sa" user, so that all team members d onot need to reset their user login every single time.

Comment (AleXLaeR) [reply .docker/.env:2]: We have changed this hardcoded password to random generated string - "JPv9bcqURjfNqzJw".

File: .docker/docker-compose.yml [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+      timeout: 5s
+      retries: 3
+    depends_on:
+      db:
+        condition: service_healthy
+    environment:
+      ASPNETCORE_ENVIRONMENT: Development
+      ASPNETCORE_URLS: "http://*:5000" # ;https://*:5001
+    volumes:
+      - "./healthchecks:/healthchecks"
```
[the remark this answers is not in the document]
Comment (AleXLaeR) [reply .docker/docker-compose.yml:35]: We are leaving it for the future scaling.

File: build/Targets/PublishBuild.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (LanchevychMaxym) [reply build/Targets/PublishBuild.cs:8]: publish usually associates with binary publish of code, but here you pushing changes to git, that's confusing

File: build/Targets/SetupPublicBuild.cs [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+partial class Build
+{
+    [Parameter("docker atom")]
+    readonly string DockerAtom = "Streetcode";
+
+    Target SetPublicEnvironmentVariables => _ => _
+        .Executes(() =>
+        {
+            //ToAsk ??? how to use and what it means
+        });
```
[the remark this answers is not in the document]
Comment (dimasster) [reply build/Targets/SetupPublicBuild.cs]: we set docker-atom name here

**Pull request merged**